### PR TITLE
[Modules] Updated Sql/Servers to new dependency approach

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Updated Sql/Servers to new dependency approach

| Pipeline |
| - |
| [Sql: Servers](https://github.com/Azure/ResourceModules/actions/workflows/ms.sql.servers.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: modules/Microsoft.Sql/servers/.test/admin/deploy.test.bicep [hunk 1]
```diff
@@ -0,0 +1,52 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for a testing purposes.')
```
Comment (MrMCake) [inline modules/Microsoft.Sql/servers/.test/admin/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes.')
```

File: modules/Microsoft.Sql/servers/.test/common/deploy.test.bicep [hunk 2]
```diff
@@ -0,0 +1,141 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for a testing purposes.')
```
Comment (MrMCake) [inline modules/Microsoft.Sql/servers/.test/common/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes.')
```

File: modules/Microsoft.Sql/servers/.test/pe/deploy.test.bicep [hunk 3]
```diff
@@ -0,0 +1,63 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for a testing purposes.')
```
Comment (MrMCake) [inline modules/Microsoft.Sql/servers/.test/pe/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes.')
```

File: modules/Microsoft.Sql/servers/.test/common/dependencies.bicep [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+@description('The principal ID of the created managed identity.')
+output managedIdentityPrincipalId string = managedIdentity.properties.principalId
+
+@description('The resource ID of the created managed identity.')
+output managedIdentitResourceId string = managedIdentity.id
+
+@description('The resource ID of the created virtual network subnet.')
+output privateEndpointSubnetResourceId string = virtualNetwork.properties.subnets[0].id
+
+@description('The resource ID of the created Virtual Network Subnet.')
```
Comment (eriqua) [inline modules/Microsoft.Sql/servers/.test/common/dependencies.bicep:61]:
```suggestion
@description('The resource ID of the created Private DNS Zone.')
```

File: modules/Microsoft.Sql/servers/.test/common/deploy.test.bicep [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+  name: resourceGroupName
+  location: location
+}
+
+module resourceGroupResources 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, location)}-nestedDependencies'
+  params: {
+    managedIdentityName: 'dep-<<namePrefix>>-msi-${serviceShort}'
+    virtualNetworkName: 'adp-<<namePrefix>>-vnet-${serviceShort}'
```
Comment (eriqua) [inline modules/Microsoft.Sql/servers/.test/common/deploy.test.bicep:36]:
```suggestion
    virtualNetworkName: 'dep-<<namePrefix>>-vnet-${serviceShort}'
```

File: modules/Microsoft.Sql/servers/.test/common/dependencies.bicep [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+  name: virtualNetworkName
+  location: location
+  properties: {
+    addressSpace: {
+      addressPrefixes: [
+        '10.0.0.0/16'
+      ]
+    }
+    subnets: [
+      {
```
Comment (eriqua) [inline modules/Microsoft.Sql/servers/.test/common/dependencies.bicep:25]: I'd suggest 2 subnets. One for the private endoint set up as required, one for the `virtualNetworkRules` (selected network) where service endpoint should be set

File: modules/Microsoft.Sql/servers/.test/common/deploy.test.bicep [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+module testDeployment '../../deploy.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name)}-test-${serviceShort}'
+  params: {
+    name: '<<namePrefix>>-${serviceShort}'
+    lock: 'CanNotDelete'
+    administratorLogin: 'adminUserName'
+    administratorLoginPassword: password
+    location: location
+    minimalTlsVersion: '1.2'
```
Comment (eriqua) [inline modules/Microsoft.Sql/servers/.test/common/deploy.test.bicep:68]: This is by default, should we remove it?

File: modules/Microsoft.Sql/servers/.test/pe/dependencies.bicep [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+      }
+      registrationEnabled: false
+    }
+  }
+}
+
+@description('The resource ID of the created virtual network subnet.')
+output subnetResourceId string = virtualNetwork.properties.subnets[0].id
+
+@description('The resource ID of the created Virtual Network Subnet.')
```
Comment (eriqua) [inline modules/Microsoft.Sql/servers/.test/pe/dependencies.bicep:47]:
```suggestion
@description('The resource ID of the created Private DNS Zone.')
```

File: modules/Microsoft.Sql/servers/.test/common/deploy.test.bicep (hunk 7 above)
Comment (MrMCake) [reply modules/Microsoft.Sql/servers/.test/common/deploy.test.bicep:68]: done

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**